### PR TITLE
[release-1.25] releng/go: Bump images, versions and deps to use Go 1.20.7 and newer protoc

### PR DESCRIPTION
**Pull request by jeremyrickard**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

- Bump images, versions and deps to use Go 1.20.7

#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/release/issues/3181

#### Does this PR introduce a user-facing change?
```release-note
Kubernetes is now built with Go 1.20.7
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs

```

/assign @saschagrunert @cpanato 
cc @kubernetes/release-engineering

### CODE REVIEW

File: hack/lib/protoc.sh [hunk 1]
```diff
@@ -29,18 +29,16 @@ function kube::protoc::generate_proto() {
   kube::golang::setup_env
   GO111MODULE=on GOPROXY=off go install k8s.io/code-generator/cmd/go-to-protobuf/protoc-gen-gogo
 
-  kube::protoc::check_protoc
```
Comment (liggitt) [inline hack/lib/protoc.sh:32]: restore this call?

**Comment by jeremyrickard** [comment]
@dims @liggitt could you review this one?

**Pull request merged**